pkg/status/singletonstatus.go

```diff
@@ -64,24 +64,41 @@ func (c *Controller) updateWorkStatusToObject(ctx context.Context, workStatusON
 func (c *Controller) syncWorkloadObject(ctx context.Context, wObjID util.ObjectIdentifier) error {
 	logger := klog.FromContext(ctx)
 	isSingletonRequested, nWECs := c.bindingPolicyResolver.GetSingletonReportedStateRequestForObject(wObjID)
-	logger.V(4).Info("Workload object (singleton status requested)", "object", wObjID, "isSingletonRequested", isSingletonRequested, "nWECs", nWECs)
 
-	// TODO: GetMultiWECReportedStateRequestForObject is yet to be implemented.
-	// just declaring these variables, we will get its actual value once above function is implemented.
-	var isMultiWECRequested bool
-	logger.V(4).Info("Workload object (multiWEC status requested)", "object", wObjID, "isMultiWECRequested", isMultiWECRequested, "nWECsMulti", nWECs)
+	isMultiWECRequested, nWECsMulti := c.bindingPolicyResolver.GetMultiWECReportedStateRequestForObject(wObjID)
 
-	if (isMultiWECRequested || isSingletonRequested) && nWECs == 1 {
-		logger.V(4).Info("Either singleton or multiWEC status is requested and nWEC == 1", "object: ", wObjID, "nWEC", nWECs)
+	// nWECs is the count of WEC matched by clusterselectors in binding policy when wantSingletonReportedState = true
+	// nWECsMulti is the count of WEC matched by clusterselectors in binding policy when wantMultiWECReportedState = true
+	// when both wantSingletonReportedState and wantMultiWECReportedState is set to true then nWECs and nWECsMulti is equal
+
+	logger.V(4).Info("Workload object (singleton status requested)", "object", wObjID, "isSingletonRequested", isSingletonRequested, "nWECs", nWECs,
+		"Workload object (multiWEC status requested)", "object", wObjID, "isMultiWECRequested", isMultiWECRequested, "nWECsMulti", nWECsMulti)
```

Comment:
Clearly nobody looked at the result of actually executing this statement.

There is no reason to duplicate the `, "object", wObjID`.

Look at the definition of the `Info` method on a Logger. See how it takes one message followed by a series of key-value pairs? `"Workload object (multiWEC status requested)"` here is a second message, which does not belong in the series of key-value pairs. Instead, generalize the one and only message at the start of this call.